javascript/javascript2/week2/movies/movies.js

```diff
@@ -0,0 +1,1729 @@
+        title: "Ah fei zing zyun",
+        year: 1990,
+        rating: 7.6,
+        votes: 13683,
+        running_times: 5640,
+    },
+    {
+        title: "Ain't Them Bodies Saints",
+        year: 2013,
+        rating: 6.4,
+        votes: 18005,
+        running_times: 5760,
+    },
+    { title: "Air", year: 2015, rating: 5.1, votes: 10524, running_times: 5700 },
+    {
+        title: "Air America",
+        year: 1990,
+        rating: 5.7,
+        votes: 23513,
+        running_times: 6780,
+    },
+    {
+        title: "Air Bud",
+        year: 1997,
+        rating: 5.1,
+        votes: 14077,
+        running_times: 5880,
+    },
+];
+
+console.log("Only " + movies.length + " movies selected");
+
+// Movies with short titles
+
+const moviesWithShortTitle = movies
+    .filter((movie) => movie.title.length < 7)
+    .map((movie) => movie.title); // gives only the title
+console.log("\nShort titled movies()\n------------");
+console.log(moviesWithShortTitle);
+
+// movies with long titles
+const moviesWithLongTitle = movies
+    .filter((movie) => movie.title.length > 20)
+    .map((movie) => movie.title);
+console.log("\nLong titled movies()\n------------");
+console.log(moviesWithLongTitle);
+
+//Movies produced between 1980 and 1989
+
+let countMoviesProduced = 0;
+const moviesOf1980s = movies.filter(
+    (movie) => movie.year >= 1980 && movie.year <= 1989
+);
+countMoviesProduced++;
+console.log("\nNumber of Movies from 1980s()\n------------");
+console.log(moviesOf1980s.length);
+
+//console.log(moviesOf1980s); // array of movies from 1980s
+
+//Adding an extra tag
+
+movies[5].key = "Tag"; //To add a new key called Tag
+movies[5].values = "Newvalue"; //To add a value pair based on rating
+
+const addingExtraTag = movies.forEach((movie) => {
+    if (movie.rating >= 7) return (movie.Tag = "Good");
+    else if (movie.rating >= 4 && movie.rating <= 7)
+        return (movie.Tag = "Medium");
+    else movie.rating < 4;
+    return (movie.Tag = "Bad");
+});
+console.log("\nAdding Extra Tag based on rating()\n-------");
+console.log(movies);
+
+//Movies rated above 6
+
+const filteringMovies = movies
+    .filter((movie) => movie.rating > 6)
+    .map((movie) => movie.rating);
+console.log("\n Movies rated above 6()\n--------------");
+console.log(filteringMovies);
+
+//Count total movies containing "keywords: Surfer, Alien or Benjamin"
+
+const moviesContainingKeywords = movies
+    .filter(
+        (movie) =>
+        movie.title.includes(`Surfer`) ||
+        movie.title.includes(`Alien`) ||
+        movie.title.includes(`Benjamin`)
+    )
+    .map((movie) => movie.title);
+
+console.log("\nMovies containing keywords()\n---------");
+console.log(moviesContainingKeywords); // Case sensitivity not working
+
+//Find deplicated word in a title
+// sytax string.match( param )
+
+const wordDiplicatedTitles = movies
+    .filter((movie) => movie.title.match(/\b(\w+)\b.*\b\1\b/g) !== null) // // ref: https://tinyurl.com/y7p7vrcj
```

Comment:
Can you think of a solution without using regular expressions? By using the string class